src/code/game.c

```diff
@@ -408,7 +408,7 @@ void GameState_Init(GameState* gameState, GameStateFunc init, GraphicsContext* g
     gameState->destroy = NULL;
     gameState->running = 1;
     startTime = osGetTime();
-    gameState->size = gameState->init = 0;
+    gameState->size = (u32)(gameState->init = NULL);
```

Comment:
That is an interesting change, is it fixing a warning?
Tbh I'd rather have the warning, there's only so much we can do about them anyway in a matching context

Reply:
This fixes a "makes integer from pointer without a cast" warning. My view is that whenever we can fix a warning we should, and I think making this line a little uglier is even a good thing since it makes the weirdness more apparent than if it was left looking cleaner.

Reply:
if we go with this id like a comment saying that this is effectively setting both to 0. that is not obvious at all when written this way

Reply:
something like `// sets 0 and NULL respectively` or something is fine with me

Reply:
Alternatively in similar chained-assignments-we-wish-were-not we have just the list of values like `0, NULL`

Reply:
possibly, but this one is espcially weird with the casting and parenthesis that make it not even look like a chained assign at first glance. it warrants a comment more than other cases imo

Reply:
Sure

Reply:
Added a comment